src/components/CarouselRulesForm/CarouselRulesForm.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react';
+import { TextInput } from '../TextInput';
+import { CarouselRules } from '../../types/CarouselRules';
+
+const INITIAL_NEW_CAROUSEL_RULES: CarouselRules = {
+  step: 0,
+  frameSize: 0,
+  itemWidth: 0,
+  animationDuration: 0,
+  infinite: false,
+};
+
+type Props = {
+  onSubmit: (newRules: CarouselRules) => void,
+};
+
+export const CarouselRulesForm: React.FC<Props> = ({
+  onSubmit,
+}) => {
+  const [newCarouselRules, setNewCarouselRules]
+    = useState(INITIAL_NEW_CAROUSEL_RULES);
+  const [formState, setFormState] = useState(0);
+
+  const [hasItemWidthError, setHasItemWidthError] = useState(false);
+  const [hasFrameSizeError, setHasFrameSizeError] = useState(false);
+  const [hasStepError, setHasStepError] = useState(false);
+  const [hasAnimationDurationError, setHasAnimationDurationError]
+    = useState(false);
+
+  const handleFieldChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const fieldName = event.target.name;
+    // eslint-disable-next-line prefer-destructuring
+    const value = +event.target.value;
+
+    setNewCarouselRules({
+      ...newCarouselRules,
+      [fieldName]: value,
+    });
+  };
+
+  const handleReset = () => {
+    setFormState(currentState => currentState + 1);
+    setNewCarouselRules(INITIAL_NEW_CAROUSEL_RULES);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasAnimationDurationError(!(newCarouselRules.animationDuration > 0));
+    setHasFrameSizeError(!(newCarouselRules.frameSize > 0));
+    setHasItemWidthError(!(newCarouselRules.itemWidth > 0));
+    setHasStepError(!(newCarouselRules.step > 0));
+
+    if (newCarouselRules.animationDuration > 0
+      && newCarouselRules.frameSize > 0
+      && newCarouselRules.itemWidth > 0
+      && newCarouselRules.step > 0
+    ) {
+      onSubmit(newCarouselRules);
+      handleReset();
+    }
+  };
+
+  const handleInfiniteInputChanged
+  = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewCarouselRules({
+      ...newCarouselRules,
+      infinite: event.target.checked,
+    });
+  };
+
+  return (
+    <form
+      className="box App__form"
+      key={formState}
+      onReset={handleReset}
+      onSubmit={(event) => handleSubmit(event)}
```

Comment:
If you pass only event to submit handler you don't need to create anonymous function
```suggestion
      onSubmit={handleSubmit}
```